.github/workflows/ci.yml

```
@@ -40,4 +40,46 @@ jobs:
       - name: verify go.mod and go.sum are consistent
         run : go mod tidy
       - name: Ensure nothing changed
-        run: git diff --exit-code
\ No newline at end of file
+        run: git diff --exit-code
+  tests:
+    name: Integration Tests
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Restore build cache
+        uses: actions/cache@v2
+        with:
+          path: ~/.cache/go-build
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.mod') }}
+      - name: Set local Go version
+        run: |
+          VERSION=`cat .go-version | awk '{printf$1}'`
+          echo "go_version=$VERSION" >> $GITHUB_ENV
+      - name: Setup Go environment
+        uses: actions/setup-go@v2
+        with:
+          go-version: "${{ env.go_version }}"
+      - name: Sync dependencies
+        run: |
+          go mod download
+          go mod tidy
+      - name: Install gotestsum
+        run: go install gotest.tools/gotestsum@latest
+      - name: Fetch Outputs
+        env:
+          TFE_TOKEN: "${{ secrets.TF_WORKFLOW_TFLOCAL_CLOUD_TFC_TOKEN }}"
+        run: |
+          go run .github/scripts/fetch_outputs/main.go hashicorp-v2 tflocal-go-tfe
+      - name: Run Tests
+        env:
+          TFE_TOKEN: "${{ env.TFE_TOKEN }}"
+          TFE_ADDRESS: "${{ env.TFE_ADDRESS }}"
+          TFC_RUN_TASK_URL: "https://httpstat.us/200"
+        run: |
+          export GO111MODULE=on
```

Non-blocker -- slightly more consistent?

```suggestion
          TFC_RUN_TASK_URL: "https://httpstat.us/200"
          GO111MODULE: "on"
        run: |
```